### ISSUE TITLE
Area selection for other monitor

### ISSUE THREAD

**Issue by eribertomota** [vkohaupt/vokoscreenNG#12]
Hi,

Currently vokoscreenNG don't allow one to select an area in other monitor. I have 3 monitors and I would like to select an area to record screen in my secondary monitor. Thanks!

Regards,

Eriberto

**Comment by eribertomota** [comment]
Hummm... I discovered that if I move the vokoscreenNG window to another monitor, it will able to select an area on this monitor only. However, I suggest to make possible keep the vokoscreenNG window in a monitor and allow the user to select an area in other monitor. Feel free to close this issue report if it is not possible.

Thanks!

**Comment by vkohaupt** [comment]
Yes, if move the vokoscreenNG window in the second monitor and select area, the area comes on this monitor but record from the first monitor. A workaround, change the second Monitor as primary screen.
The area-gui is not a simple programming thing, it works under X11, Windows and Wayland. I think that I will extend the area with version 4.0.

**Comment by eribertomota** [comment]
In KDE I edited .config/vokoscreenNG/vokoscreenNG.ini and I set in [AreaScreencast] the X and Y values to record in second screen. It worked but is hard to users.

Thanks for your reply.

**Comment by Martii** [comment]
Don't quote me on this but when I was testing vokoscreenNG I encountered the same issue. What I think I did was started vokoscreenNG up on a specific monitor and that was the screen where the selection rectangle was... I think... it's been a few months since I did that and I'm not system ready to recompile it atm due to SSD failure *(again)*.

Anyways... give that a shot. It might work and could be a little easier than .ini editing.

**Comment by eribertomota** [comment]
Hi @vkohaupt!

Any news? I need this resource to remove the old Vokoscreen from Debian.

Thanks!

**Comment by vkohaupt** [comment]
What does the old have to do with the new?

**Comment by eribertomota** [comment]
This feature works fine in old version. I and other users need it.

**Comment by vkohaupt** [comment]
I only have one screen so I never tested it. But I have experienced that Virtualbox supports virtual screens. Let's see if I can handle it. Please be patient and I cannot promise it.

**Comment by klaaslusterbroodvan** [comment]
When will Wayland support come for Linux? For example, I use Gnome 3.36 on Fedora 32.

I prefer Wayland because some applications (in particular Firefox) look much better on Wayland than on X11, and I want to make tutorial videos using the best looking software.
Many thanks for the hard work on this great software package.

**Comment by vkohaupt** [comment]
@eribertomota
Area supports now multiscreen.
Please test it.

**Comment by eribertomota** [comment]
Hi @vkohaupt 

Now, when I move the main windown of Vokoscreen to another screen and select Area, the box to select an area is shown in right video monitor. However, since some versions ago the content is truncated. See here[1]. Now, see here[2] the same content in Fullscreen mode. The window mode works too.

[1] http://eriberto.pro.br/files/tmp/vokoscreenNG-2020-04-12_22-07-02.mkv
[2] http://eriberto.pro.br/files/tmp/vokoscreenNG-2020-04-12_22-09-39.mkv

I already removed all previous configuration file ($ rm -rf ~/.config/vokoscreenNG/)

Thanks in advance.

Eriberto

**Comment by vkohaupt** [comment]
Sorry my fault. You need to compile from master. I commited the bugfix yesterday.

**Comment by eribertomota** [comment]
Hi @vkohaupt,

No fault. I compiled from master. The 3.0.3 version already in Debian since 2020-04-06, so I used master for my last message.

Tested 6a90ce6 over Debian 10, kernel 4.19.98-1, KDE. Let me know if you need more details or tests.

Thanks!

Eriberto

**Comment by eribertomota** [comment]
I tested up to 33d4ba6 now and Area is working fine! However, now, Window records nothing. Full screen is ok.

**Comment by vkohaupt** [comment]
Window records nothing? Hmmm....
Is that always the case? Or just now and then.
Is the Video black or the size is zero?
Please make a short Video and post the complete log here.

**Comment by eribertomota** [comment]
Is intermittent for me. I saw this behaviour in 6a90ce6 and now, but I don't know if something was wrong in past versions. If it is relevant I can try test with 3.0.3, 3.0.2... But it is intermittent.

Some moments after...

I detected now the problem is in Play mode only. If I close vokoscreenNG and open with vlc, the video works.

i will close this issue and open a new issue about it if needed. When you release the 3.0.4, I will make new tests, upload to Debian and remove the old vokoscreen from Debian.

Thanks a lot for your work.

Cheers,

Eriberto
